paywall/src/data-iframe/blockchainHandler/pollForChanges.js

```diff
@@ -0,0 +1,48 @@
+import { delayPromise } from '../../utils/promises'
+
+/**
+ * Compare 2 things and return truthy if they are different
+ *
+ * @callback compareFunc
+ * @param {*} oldValue
+ * @param {*} newValue
+ * @returns {bool} true if the values are different
+ */
+
+/**
+ * @callback changeListener
+ * @param {*} newValue
+ */
+
+/**
+ * @callback continuePolling
+ * @param {*} currentValue
+ * @returns {bool} true if the polling should continue
+ */
+
+/**
+ * Poll for changes every 5 seconds in something, and call the changeListener when it changes
+ *
+ * @param {Function} getCurrentValue should return the current value(s) we are polling for changes
+ * @param {compareFunc} hasValueChanged
+ * @param {continuePolling} continuePolling
+ * @param {changeListener} changeListener
+ * @param {int} delay
+ */
+export default async function pollForChanges(
+  getCurrentValue,
+  hasValueChanged,
+  continuePolling,
+  changeListener,
+  delay
+) {
+  let before = await getCurrentValue()
+  while (await continuePolling(before)) {
+    await delayPromise(delay)
+    const after = await getCurrentValue()
+    if (await hasValueChanged(before, after)) {
```

Comment:
Does this need to be async? I would assume that the before and after values are actually resolved and then this can be sync?

Reply:
In most cases, no it doesn't. However, `await` is not just for async, you can await any value and it will just pretend you wrapped it with `Promise.resolve()` (see the end of https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/await#Description). So, by adding the `await` before this, we lose nothing, but gain flexibility should there be a use case where we need to do some async processing to do the comparison.